src/widgets/barchart.rs

```diff
@@ -371,35 +388,25 @@ impl<'a> BarChart<'a> {
             .max()
             .unwrap_or(0) as u16;
 
-        let label_x = bars_area.x;
+        let label_x = area.x;
         let bars_area = {
             let margin = if label_size == 0 { 0 } else { 1 };
             Rect {
-                x: bars_area.x + label_size + margin,
-                width: bars_area.width - label_size - margin,
-                ..bars_area
+                x: area.x + label_size + margin,
+                width: area.width - label_size - margin,
+                ..area
             }
         };
 
-        // convert the bar values to ratatui::symbols::bar::Set
-        let groups: Vec<Vec<u16>> = self
-            .data
-            .iter()
-            .map(|group| {
-                group
-                    .bars
-                    .iter()
-                    .map(|bar| (bar.value * u64::from(bars_area.width) / max) as u16)
-                    .collect()
-            })
-            .collect();
+        let group_ticks = self.group_ticks(bars_area.height, bars_area.width);
 
         // print all visible bars, label and values
         let mut bar_y = bars_area.top();
-        for (group_data, mut group) in groups.into_iter().zip(self.data) {
+        for (ticks_vec, mut group) in group_ticks.into_iter().zip(self.data) {
             let bars = std::mem::take(&mut group.bars);
 
-            for (bar_length, bar) in group_data.into_iter().zip(bars) {
+            for (ticks, bar) in ticks_vec.into_iter().zip(bars) {
+                let bar_length = (ticks >> 3) as u16;
```

Comment:
Prefer this for clarity unless there's a measured performance difference (I'd guess this is probably automatically compiler optimized)
```suggestion
                let bar_length = (ticks /8 ) as u16;
```

Reply:
yes. I don't think that a slightly slower cpu instruction can be measured in this context.
done